xonsh/tools.py

```diff
@@ -583,7 +572,7 @@ def ensure_int_or_slice(x):
     """Makes sure that x is list-indexable."""
     if x is None:
         return slice(None)
-    elif is_int(x):
+    elif isinstance(x, int):
```

Comment:
Well, if we ever need to change `is_int()` for whatever reason, that change will be propogated here. That was my only concern. 

>  i'm currently working on a patch that allows more versatile to define ensurer functions and constraints.

Cool! I look forward to it.